mlops_utilities/actions.py

```diff
@@ -177,28 +191,39 @@ def compare_metrics(sagemaker_client,
     :return: result of metric comparison
     """
     model_deployed_description = sagemaker_client.describe_model(
-        ModelName=endpoint_config_description["ProductionVariants"][0]["ModelName"])
+        ModelName=endpoint_config_description["ProductionVariants"][0]["ModelName"]
+    )
     model_deployed_description = sagemaker_client.describe_model_package(
-        ModelPackageName=model_deployed_description["Containers"][0]["ModelPackageName"])
+        ModelPackageName=model_deployed_description["Containers"][0]["ModelPackageName"]
+    )
     test_json_metrics = {"regression_metrics": {"mse": {"value": 4}}}
```

Comment:
IMHO, I don't like the idea to complicate the function for the sake of making it testable and passing all tests. We must not mock anything in functions

Reply:
Not to be fixed in this PR. Remember, this PR is only for "refactoring"

Reply:
Again, this approach is used throughout this whole file 🫤

Reply:
Agree, revert it back, and apply only refactoring changes if any